src/lib.rs

```diff
@@ -292,44 +295,89 @@ impl Wallet {
     }
 }
 
-pub struct ExtendedKeyInfo {
+pub struct DescriptorKeyInfo {
+    xprv: Option<String>,
+    xpub: String,
+}
```

Comment:
welcome this type! @notmandatory do you have any take on the name?

Reply:
Now that I'm looking at it all together and the naming I'm thinking my original suggestion is not quite right. The use cases I'm trying to solve for are:

    Generate new mnemonic seed words, or create from existing seed words
    Create new extended private key for a mnemonic and network, with optional password
    Derive new public or private extended key from a private extended key and derivation path
    (future) Create a wallet descriptor from a standard template given a public or private extended key

Based on this I think the types we need are Mnemonic, ExtendedPrivKey, ExtendedPubKey.